mysql-test/main/sp.test

```diff
@@ -10661,11 +10661,6 @@ CREATE OR REPLACE FUNCTION ordinality(a VARCHAR(128)) RETURNS int RETURN LENGTH(
 SELECT ordinality('1');
 DROP FUNCTION ordinality;
 
-CREATE OR REPLACE FUNCTION path(a VARCHAR(128)) RETURNS int RETURN LENGTH(a)=0;
```

Comment:
the purpose here is about return values or something unrelated to "path". Just rename the function in the test to something that doesn't conflict.

More seriously though. is this adding PATH as a keyword that needs to be avoided in naming?

Reply:
Regarding the concern,
`is this adding PATH as a keyword that needs to be avoided in naming`
I'm also not sure. 
Maybe @abarkov or @FooBarrior know more about this?

However, I found that there is a similar set command,
which is [SET ROLE](https://mariadb.com/kb/en/set-role/)
also prohibit the user to create the routine name.

Note that In grammar file, sql_yacc.yy, I have moved the PATH symbol (PATH_SYM)
to nonterminal name, `keyword_set_special_case`,
from nonterminal name, `keyword_func_sp_var_and_label`.
If I don't do so, I will face the build error that
`mariadb/build-mariadb-server-debug/sql/yy_mariadb.yy: error: reduce/reduce conflicts: 1 found, 0 expected`

As a result, the routine name is not able to create with the name `path` in the unit test, sp.test.

Reply:
Do you guys have any hints for me that how to solve the grammar's reduce conflict build failed
in order not to change the unit test, sp.test?

Reply:
This patch seems to fix the problem:
```
diff --git a/sql/sql_yacc.yy b/sql/sql_yacc.yy
index b006eed628c..b917d6a0bd3 100644
--- a/sql/sql_yacc.yy
+++ b/sql/sql_yacc.yy
@@ -10713,14 +10713,6 @@ function_call_conflict:
             if (unlikely($$ == NULL))
               MYSQL_YYABORT;
           }
-        | PASSWORD_SYM '(' expr ')'
-          {
-            Item* i1;
-            i1= new (thd->mem_root) Item_func_password(thd, $3);
-            if (unlikely(i1 == NULL))
-              MYSQL_YYABORT;
-            $$= i1;
-          }
         | REPEAT_SYM '(' expr ',' expr ')'
           {
             $$= new (thd->mem_root) Item_func_repeat(thd, $3, $5);
@@ -15825,6 +15817,7 @@ IDENT_sys:
 ident_cli_func:
           IDENT
         | IDENT_QUOTED
+        | keyword_set_special_case       { $$= $1; }
         | keyword_func_sp_var_and_label  { $$= $1; }
         | keyword_func_sp_var_not_label  { $$= $1; }
         ;

```

Reply:
Hi @abarkov ,

Thank you so much of your patch!
It works and also without side effect.
Really appreciate of your help!

Reply:
Apply the patch from @abarkov .